app/code/Magento/Catalog/Model/CategoryRepository.php

```diff
@@ -239,4 +236,25 @@ private function getMetadataPool()
         }
         return $this->metadataPool;
     }
+
+    /**
+     * Get existing data category
+     *
+     * @param CategoryInterface $category
+     * @param int $storeId
+     * @return array
+     */
+    private function getExistingData(CategoryInterface $category, int $storeId)
+    {
+        $existingData = $this->getExtensibleDataObjectConverter()
+            ->toNestedArray($category, [], \Magento\Catalog\Api\Data\CategoryInterface::class);
+        $existingData = array_diff_key($existingData, array_flip(['path', 'level', 'parent_id']));
+        $existingData['store_id'] = $storeId;
+
+        if ($category->getData('save_rewrites_history') !== null) {
+            $existingData['save_rewrites_history'] = $category->getData('save_rewrites_history');
+        }
```

Comment:
This isn't ideal, as mentioned by @swnsma it leads to violation of modularity, where `Catalog` module knows about an attribute introduced by `CatalogUrlRewrite` which isn't ideal. 

Instead of changing the repository itself, can we create a plugin (just on rest webapi context, similar to `CategoryInputParamsResolver`) that runs before `CategoryUrlRewriteGenerator::generate` and check the RestRequest for the `save_rewrites_history` property and sets it on the `Category` object?